lib/elixir/lib/kernel/typespec.ex

```diff
@@ -99,8 +99,9 @@ defmodule Kernel.Typespec do
   Invoked by `Kernel.@/1` expansion.
   """
   def deftypespec(:spec, expr, _line, _file, module, pos) do
-    {_set, bag} = :elixir_module.data_tables(module)
+    {set, bag} = :elixir_module.data_tables(module)
     store_typespec(bag, :spec, {expr, pos})
+    store_module_attribute(set, :spec, {expr, pos})
```

Comment:
How did it work on Elixir v1.7? Did `@spec` return only the last `@spec` or would it return all of them?

Reply:
It accumulates, returning all the defined ones until that point.

I also notices it includes everything into a `{:spec, ...}` tuple.

(I guess you meant v1.6, because v1.7 does not allow to use `@spec` as module attribute)

Reply:
Yes, v1.6. :) We should mirror the return type there. I am wondering if instead if we should change `Module.get_attribute` and make it return the old format for `@spec`. We should also probably forbid `Module.put_attribute` with a spec (if we haven't yet).

Reply:
It sounds good to me!

Should `Module.get_attribute` be the one responsible for returning the old format, or should we store the old format directly?

Reply:
I would change Module.get_attribute to compute the old format. Very few projects actually want to read it. So I would "punish" the readers rather than the writers.